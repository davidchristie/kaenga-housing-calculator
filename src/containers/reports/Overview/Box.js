import React, { Component } from 'react'
import { Panel } from 'react-bootstrap'

export default class Box extends Component {
  render () {
    return (
      <Panel
        style={{
          fontSize: '1.6em',
          textAlign: 'center'
        }}
      >
        {this.props.children}
      </Panel>
    )
  }
}
