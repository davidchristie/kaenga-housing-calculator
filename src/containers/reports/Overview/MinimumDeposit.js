import React, { Component } from 'react'
import { Panel } from 'react-bootstrap'
import { connect } from 'react-redux'

import Currency from '../../../components/units/Currency'
import minimumDeposit from '../../../models/minimum-deposit'

class MinimumDeposit extends Component {
  render () {
    return (
      <Panel
        header='Minimum Deposit'
        bsStyle='primary'
        style={{
          fontSize: '1.6em',
          textAlign: 'center'
        }}
      >
        <strong>
          <Currency value={this.props.value} />
        </strong>
      </Panel>
    )
  }
}

export default connect(
  state => {
    return {
      value: minimumDeposit(state)
    }
  }
)(MinimumDeposit)
