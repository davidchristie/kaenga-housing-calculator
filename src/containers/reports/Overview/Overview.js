import React from 'react'
import { connect } from 'react-redux'
import { Col, Row, Well } from 'react-bootstrap'

import design from '../../../selectors/design'
import housingType from '../../../selectors/housing-type'
import PriceRangeWithSuburbs from '../../PriceRangeWithSuburbs'
import SavingsVsAverageHouse from '../../SavingsVsAverageHouse'
import YearlyCost from '../../YearlyCost'

class Overview extends React.Component {
  render () {
    return (
      <Well>
        <h2>Overview</h2>
        <Row>
          <Col md={6}>
            <p>
              Price: <PriceRangeWithSuburbs />
            </p>
            <p>
              House Type: <strong>{this.props.houseType}</strong>
            </p>
            <p>
              Bedrooms: <strong>{this.props.bedrooms}</strong>
            </p>
          </Col>
          <Col md={6}>
            <p>
              Savings verse average house:
              <strong> <SavingsVsAverageHouse /></strong>
            </p>
            <p>
              Title: <strong>Freehold Unit Title</strong>
            </p>
            <p>
              Bathrooms: <strong>{this.props.bathrooms}</strong>
            </p>
          </Col>
        </ Row>
        <YearlyCost />
      </Well>
    )
  }
}

export default connect(
  state => {
    return {
      bathrooms: design(state).bathrooms || 0,
      bedrooms: design(state).bedrooms || 0,
      houseType: housingType(state).type || ''
    }
  }
)(Overview)
