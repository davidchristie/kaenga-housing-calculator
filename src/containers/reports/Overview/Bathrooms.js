import React, { Component } from 'react'
import { connect } from 'react-redux'

import design from '../../../selectors/design'
import Box from './Box'

class Bathrooms extends Component {
  render () {
    return (
      <Box>
        <h4>Bathrooms</h4>
        <strong>{this.props.value}</strong>
      </Box>
    )
  }
}

export default connect(
  state => {
    return {
      value: design(state).bathrooms
    }
  }
)(Bathrooms)
