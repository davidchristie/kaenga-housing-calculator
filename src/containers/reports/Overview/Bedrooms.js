import React, { Component } from 'react'
import { connect } from 'react-redux'

import design from '../../../selectors/design'
import Box from './Box'

class Bedrooms extends Component {
  render () {
    return (
      <Box>
        <h4>Bedrooms</h4>
        <strong>{this.props.value}</strong>
      </Box>
    )
  }
}

export default connect(
  state => {
    return {
      value: design(state).bedrooms
    }
  }
)(Bedrooms)
