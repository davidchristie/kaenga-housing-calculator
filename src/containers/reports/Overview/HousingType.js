import React, { Component } from 'react'
import { connect } from 'react-redux'

import housingType from '../../../selectors/housing-type'
import Box from './Box'

class HousingType extends Component {
  render () {
    const { values } = this.props
    const terraced = values.terracedHousing
    const apartment = Math.max(
      values.highRise,
      values.midScaleApartment,
      values.largeScaleApartment
    )
    let type = null
    if (terraced === apartment) {
      type = 'Terraced housing or apartment'
    } else if (terraced > apartment) {
      type = 'Terraced housing'
    } else {
      type = 'Apartment'
    }
    return (
      <Box>
        <h4>Housing Type</h4>
        <strong>{type}</strong>
      </Box>
    )
  }
}

export default connect(
  state => {
    return {
      values: housingType(state)
    }
  }
)(HousingType)
