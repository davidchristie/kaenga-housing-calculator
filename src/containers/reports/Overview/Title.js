import React, { Component } from 'react'

import Box from './Box'

export default class Title extends Component {
  render () {
    return (
      <Box>
        <h4>Title</h4>
        <strong>
          Freehold Unit Title
        </strong>
      </Box>
    )
  }
}
