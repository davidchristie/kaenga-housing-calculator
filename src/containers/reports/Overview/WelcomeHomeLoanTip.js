import React, { Component } from 'react'
import { connect } from 'react-redux'

import isEligibleForWelcomeHomeLoan from '../../../models/isEligibleForWelcomeHomeLoan'
import Box from './Box'

class WelcomeHomeLoanTip extends Component {
  render () {
    const style = this.props.show ? {} : {display: 'none'}
    return (
      <Box style={style}>
        If you earn less than $85,000 as an individual or $130,000 as a couple, you may be eligible for a Welcome Home Loan which would mean you only need a 10% deposit. <a href='http://www.welcomehomeloan.co.nz/am-i-eligible' target='_blank'>Click here</a> for details.
      </Box>
    )
  }
}

export default connect(
  state => {
    return {
      show: isEligibleForWelcomeHomeLoan(state)
    }
  }
)(WelcomeHomeLoanTip)
