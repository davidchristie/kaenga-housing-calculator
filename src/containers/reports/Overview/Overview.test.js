import React from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import Overview from './Overview'

const mockStore = configureStore()

it('matches snapshot', () => {
  const store = mockStore({
    suburbs: []
  })
  const tree = renderer.create(
    <IntlProvider locale='en'>
      <Provider store={store}>
        <Overview />
      </Provider>
    </IntlProvider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
