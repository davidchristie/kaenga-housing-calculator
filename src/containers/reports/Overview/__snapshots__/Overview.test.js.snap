exports[`test matches snapshot 1`] = `
<div
  className="well">
  <h2>
    Overview
  </h2>
  <div
    className="row">
    <div
      className="col-md-6">
      <p>
        Price: 
        <span />
      </p>
      <p>
        House Type: 
        <strong>
          
        </strong>
      </p>
      <p>
        Bedrooms: 
        <strong>
          0
        </strong>
      </p>
    </div>
    <div
      className="col-md-6">
      <p>
        Savings verse average house:
        <strong>
           
          <span>
            <span>
              $
              <span>
                0
              </span>
            </span>
             - 
            <span>
              <span>
                0
              </span>
              %
            </span>
          </span>
        </strong>
      </p>
      <p>
        Title: 
        <strong>
          Freehold Unit Title
        </strong>
      </p>
      <p>
        Bathrooms: 
        <strong>
          0
        </strong>
      </p>
    </div>
  </div>
  <div
    className="well">
    <h2>
      Yearly Cost
    </h2>
    <h3>
      Electricity
    </h3>
    <p>
      Total: 
      <span>
        $
        <span>
          0
        </span>
      </span>
    </p>
    <p>
      Saving: 
      <span>
        $
        <span>
          3.1
        </span>
        K
      </span>
       - 
      <span>
        <span>
          100
        </span>
        %
      </span>
    </p>
  </div>
</div>
`;
